### PR TITLE
Show the reason why an add-on is not available

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

Show the reason why an add-on is not available

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [:joy:] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: hassio/src/addon-view/info/hassio-addon-info.ts [hunk 1]
```diff
@@ -170,11 +173,25 @@ class HassioAddonInfo extends LitElement {
...
-                        This update is no longer compatible with your system.
-                      </p>
-                    `
+                  ? !this.addon.arch.find((arch) =>
+                      this.supervisor.info.supported_arch.includes(arch)
+                    )
+                    ? html`
+                        <p>
+                          The architecture on your system
+                          (${this.supervisor.supervisor.arch}) is no longer
```
Comment (frenck) [inline hassio/src/addon-view/info/hassio-addon-info.ts:182]: We don't know if it is "no longer", or "never has been".

```suggestion
                          The architecture on your system
                          (${this.supervisor.supervisor.arch}) is not
```

Comment (ludeeus) [reply hassio/src/addon-view/info/hassio-addon-info.ts:182]: So not display archs at all?

File: hassio/src/addon-view/info/hassio-addon-info.ts [hunk 2]
```diff
@@ -170,11 +173,24 @@ class HassioAddonInfo extends LitElement {
                   iconClass="update"
                 ></hassio-card-content>
                 ${!this.addon.available
-                  ? html`
-                      <p>
-                        This update is no longer compatible with your system.
-                      </p>
-                    `
+                  ? !this.addon.arch.find((arch) =>
```
Comment (bramkragten) [inline hassio/src/addon-view/info/hassio-addon-info.ts:176]:
```suggestion
                  ? !this.addon.arch.some((arch) =>
```

File: hassio/src/addon-view/info/hassio-addon-info.ts [hunk 3]
```diff
@@ -600,21 +616,32 @@ class HassioAddonInfo extends LitElement {
...
-                  : ""}
-                <ha-progress-button
-                  .disabled=${!this.addon.available}
-                  @click=${this._installClicked}
-                >
-                  Install
-                </ha-progress-button>
-              `}
+            : !this.addon.available
+            ? !this.addon.arch.find((arch) =>
```
Comment (bramkragten) [inline hassio/src/addon-view/info/hassio-addon-info.ts:620]:
```suggestion
            ? !this.addon.arch.some((arch) =>
```

File: hassio/src/addon-view/info/hassio-addon-info.ts [hunk 4]
```diff
@@ -170,11 +173,24 @@ class HassioAddonInfo extends LitElement {
...
                 ></hassio-card-content>
                 ${!this.addon.available
-                  ? html`
-                      <p>
-                        This update is no longer compatible with your system.
-                      </p>
-                    `
+                  ? !this.addon.arch.some((arch) =>
+                      this.supervisor.info.supported_arch.includes(arch)
+                    )
```
Comment (bramkragten) [inline hassio/src/addon-view/info/hassio-addon-info.ts:178]: Might want to create a helper for this? 🤷‍♂️

**Review by bramkragten — APPROVED** [review]

File: src/data/hassio/supervisor.ts [hunk 5]
```diff
@@ -51,7 +51,7 @@ export type HassioInfo = {
   operating_system: string;
   supervisor: string;
   supported: boolean;
-  supported_arch: string[];
+  supported_arch: ["armhf" | "armv7" | "aarch64" | "i386" | "amd64"];
```
Comment (bramkragten) [inline src/data/hassio/supervisor.ts:54]: create a type for the archs, so you don't have to update them everywhere all the time

File: src/data/supervisor/supervisor.ts [hunk 6]
```diff
@@ -7,6 +7,10 @@ import {
   HassioSupervisorInfo,
 } from "../hassio/supervisor";
 
+export type SupervisorArchs = [
+  "armhf" | "armv7" | "aarch64" | "i386" | "amd64"
+];
```
Comment (bramkragten) [inline src/data/supervisor/supervisor.ts:12]:
```suggestion
export type SupervisorArchs = "armhf" | "armv7" | "aarch64" | "i386" | "amd64";
```

File: src/data/hassio/supervisor.ts [hunk 7]
```diff
@@ -22,7 +23,7 @@ export type HassioHomeAssistantInfo = {
 export type HassioSupervisorInfo = {
   addons: HassioAddonInfo[];
   addons_repositories: HassioAddonRepository[];
-  arch: string;
+  arch: "armhf" | "armv7" | "aarch64" | "i386" | "amd64";
```
Comment (bramkragten) [inline src/data/hassio/supervisor.ts:26]:
```suggestion
  arch: SupervisorArchs;
```

File: src/data/supervisor/supervisor.ts [hunk 8]
```diff
@@ -7,6 +7,8 @@ import {
   HassioSupervisorInfo,
 } from "../hassio/supervisor";
 
+export type SupervisorArchs = "armhf" | "armv7" | "aarch64" | "i386" | "amd64";
```
Comment (bramkragten) [inline src/data/supervisor/supervisor.ts:10]:
```suggestion
export type SupervisorArch = "armhf" | "armv7" | "aarch64" | "i386" | "amd64";
```

**Review by bramkragten — APPROVED** [review]

**Pull request merged**